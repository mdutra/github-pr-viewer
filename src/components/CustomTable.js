import React from 'react';
import PropTypes from 'prop-types';
import MaterialTable from 'material-table';

function CustomTable({
  repo,
  data,
  isLoading,
  pageSize,
  onChangePage,
  emptyDataSourceMessage,
}) {
  const fromRepo = repo ? ` from ${repo}` : '';

  return (
    <MaterialTable
      title={'Open pull requests' + fromRepo}
      columns={[{ field: 'title' }]}
      data={data}
      isLoading={isLoading}
      options={{
        search: false,
        header: false,
        pageSize,
        pageSizeOptions: [],
        showFirstLastPageButtons: false,
      }}
      localization={{
        pagination: {
          labelDisplayedRows: '{from}--{to}',
        },
        body: { emptyDataSourceMessage },
      }}
      onChangePage={onChangePage}
    />
  );
}

CustomTable.propTypes = {
  repo: PropTypes.string,
  data: PropTypes.arrayOf(
    PropTypes.shape({ title: PropTypes.string.isRequired }),
  ),
  isLoading: PropTypes.bool,
  pageSize: PropTypes.number,
  onChangePage: PropTypes.func,
  emptyDataSourceMessage: PropTypes.string,
};

CustomTable.defaultProps = {
  repo: undefined,
  isLoading: false,
  pageSize: 5,
  onChangePage: undefined,
  emptyDataSourceMessage: 'No records to display',
};

export default CustomTable;
