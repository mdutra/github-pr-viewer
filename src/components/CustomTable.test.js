import React from 'react';
import { render, cleanup } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import ShallowRenderer from 'react-test-renderer/shallow';

import CustomTable from './CustomTable';

afterEach(cleanup);

it('displays the title', () => {
  const repo = 'user/repo';

  const { container } = render(<CustomTable repo={repo} />);

  expect(container).toHaveTextContent('Open pull requests from ' + repo);
});

it('displays row data', () => {
  const data = [
    { title: 'One pull request' },
    { title: 'Another pull request' },
  ];

  const { container } = render(<CustomTable data={data} />);

  data.forEach(({ title }) => {
    expect(container).toHaveTextContent(title);
  });
});

it("shows message when there's no row data", () => {
  const message = 'No records to display';

  let { container } = render(<CustomTable />);

  expect(container).toHaveTextContent(message);

  ({ container } = render(<CustomTable data={[]} />));

  expect(container).toHaveTextContent(message);
});

it('matches snapshot', () => {
  const renderer = new ShallowRenderer();
  const repo = 'user/repo';

  const result = renderer.render(<CustomTable repo={repo} />);

  expect(result).toMatchSnapshot();
});
