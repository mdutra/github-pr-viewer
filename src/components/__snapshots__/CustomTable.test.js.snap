// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<ForwardRef(WithStyles)
  columns={
    Array [
      Object {
        "field": "title",
      },
    ]
  }
  isLoading={false}
  localization={
    Object {
      "body": Object {
        "emptyDataSourceMessage": "No records to display",
      },
      "pagination": Object {
        "labelDisplayedRows": "{from}--{to}",
      },
    }
  }
  options={
    Object {
      "header": false,
      "pageSize": 5,
      "pageSizeOptions": Array [],
      "search": false,
      "showFirstLastPageButtons": false,
    }
  }
  title="Open pull requests from user/repo"
/>
`;
